homeassistant/__main__.py

```diff
@@ -19,15 +19,19 @@
 )
 
 
-def attempt_use_uvloop() -> None:
+def set_loop() -> None:
     """Attempt to use uvloop."""
     import asyncio
-    try:
-        import uvloop
-    except ImportError:
-        pass
+
+    if sys.platform == 'win32':
+        asyncio.set_event_loop(asyncio.ProactorEventLoop())
     else:
-        asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
+        try:
+            import uvloop
+        except ImportError:
+            pass
+        else:
+            asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
```

Comment:
Could be simple:
```
try:
   import uvloop
   asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
except ..:
  pass
```

Reply:
Yes it's simpler, but the part that's supposed to raise an `ImportError` is the actual `import`. I don't want to guard the policy call because if `uvloop` has an internal problem which happens to raise an `ImportError` the system may be left in an inconsistent state and the error won't be reported.

All of this is probably not relevant in this particular case, but I got into the habit of following this pattern as a matter of principle because it's good coding practice and I did hit that set of circumstances often enough to be wary of it.